tests/test_samples.py

```diff
@@ -0,0 +1,283 @@
+                    if target != None and replacement != None:
+                        if line.startswith(target):
+                            line = replacement
+                    new_sample_file.write(line)
+
+        if not full_copy:
+            os.remove(base_filename)
+            os.rename(target_filename, base_filename)
+            os.chmod(base_filename, 0o777)
+
+
+    def __del__(self):
+        self.temp_file.close()
+        os.remove(self._temp_file.name)
+
+
+class TestSamples(BaseClientTest):
+
+    SAMPLE_FOLDER = str(os.path.dirname(
+        os.path.dirname(
+            os.path.abspath(__file__)
+        )
+    ).replace("\\", "/")) + "/sample"
+
+    BASIC_FOLDER = SAMPLE_FOLDER + "/basic"
+    ADVANCED_FOLDER = SAMPLE_FOLDER + "/advanced"
+
+
+    def test_basicfirstref_example(self):
+        # Modify sample file to include necessary sample data
+        sample_filename = self.BASIC_FOLDER + "/basic_first_ref_example.py"
+        temp_sample_file = TempSampleFile(sample_filename)
+
+        target_line = "FILE_MD5 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.MD5] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        target_line = "FILE_SHA1 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.SHA1] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        target_line = "FILE_SHA256 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.SHA256] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            dxl_client.connect()
+
+            with MockTieServer(dxl_client):
+                mock_print = run_sample(temp_sample_file.temp_file.name)
+
+                mock_print.assert_any_call(
+                    StringContains("Systems that have referenced the file:")
+                )
+
+                mock_print.assert_any_call(
+                    StringDoesNotContain("Error")
+                )
+
+            dxl_client.disconnect()
+
+
+    def test_basicgetrep_example(self):
+        # Modify sample file to include necessary sample data
+        sample_filename = self.BASIC_FOLDER + "/basic_get_reputation_example.py"
+        temp_sample_file = TempSampleFile(sample_filename)
+
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            dxl_client.connect()
+
+            with MockTieServer(dxl_client):
+                mock_print = run_sample(temp_sample_file.temp_file.name)
+
+                mock_print.assert_any_call(
+                    StringContains("\"1451502875\"")
```

Comment:
Might be nice to validate a bit more of the response content here - like the `providerId` and `trustLevel` for at least one of the reputation providers?

Reply:
Done!